### PR TITLE
Add gemfiles for rails 4.x

### LINKED ISSUE #47: get this gem working with 4.x

### PR DESCRIPTION
**Pull request by iffyuva**
fixes #47

### CODE REVIEW

File: lib/invoicing/ledger_item.rb [hunk 1]
```diff
@@ -387,7 +387,8 @@ def acts_as_payment(options={})
 
       scope :exclude_empty_invoices, lambda {
         line_items_assoc_id = ledger_item_class_info.method(:line_items).to_sym
-        line_items_refl = reflections[line_items_assoc_id]
+        line_items_refl = reflections[line_items_assoc_id] ||
```
Comment (deepak) [inline lib/invoicing/ledger_item.rb]: why is this required ?

File: lib/invoicing/taxable.rb [hunk 2]
```diff
@@ -250,7 +250,12 @@ def write_attribute(attribute, value) #:nodoc:
         @taxed_or_untaxed[attribute] = :untaxed
         @taxed_attributes[attribute] = nil
       elsif attribute =~ /^(#{attr_regex})_taxed$/
-        @attributes[attribute] = value
+        if ActiveRecord::VERSION::MAJOR == 4 && ActiveRecord::VERSION::MINOR >= 2
```
Comment (deepak) [inline lib/invoicing/taxable.rb]: any better way ?

**Comment by deepak** [comment]
do not understand how & where `FlexMock` was used for ?
can you explain and how it was removed ?
was used only for mocking database connection ?

File: lib/invoicing/taxable.rb (hunk 2 above)
Comment (iffyuva) [inline lib/invoicing/taxable.rb]: searched if Rails provides some methods, but couldn't find method which returns major and minor versions.

**Comment by iffyuva** [comment]
this gem provides support for caching AR queries. ie, with older versions of rails, if we do `TaxRate.all`, it will fetch tax rates from db for the first time, and for subsequent `TaxRate.all` queries, it will return records from the cache it maintains. But, since rails 3.2 +, and 4+ support caching query results out of box, i dont think we should support caching records feature. `FlexMock` was used to test this feature.

File: lib/invoicing/ledger_item.rb (hunk 1 above)
Comment (iffyuva) [inline lib/invoicing/ledger_item.rb]: rails 4.2 is not returning `HashWithIndifferentAccess` anymore (for reflections). so, i did a quick fix. this needs cleanup though.

File: lib/invoicing/taxable.rb (hunk 2 above)
Comment (deepak) [inline lib/invoicing/taxable.rb]: can use something like

```ruby
ActiveRecord.version >= Gem::Version.new('4.2.0')
```

Comment (deepak) [inline lib/invoicing/taxable.rb]: you have a equality for version 4. this is `>=` 
slightly different
